// @framer/gaearon.delay/code/Delay.tsx
import * as React from "react"
import { Frame, PropertyControls, ControlType, Animatable } from "framer"

interface Props {
    indicatorColor: string
    width: number
    height: number
    delay: number
}

export class Delay extends React.Component<Partial<Props>> {
    static defaultProps = {
        indicatorColor: "rgba(142,142,147,100)",
        width: 200,
        height: 200,
        delay: 1,
    }

    lastTick = performance.now()
    rotation = Animatable(0)
    state = {ready: false}

    static propertyControls: PropertyControls<Props> = {
        indicatorColor: { type: ControlType.Color, title: "Indicator" },
        delay: { type: ControlType.Number, min: 0, max: 5, title: "Seconds" },
    }

    rafHandle: number

    componentDidMount() {
        this.tick()
        this.timeout = setTimeout(() => {
          this.setState({ ready: true });
        }, this.props.delay * 1000);
    }

    componentWillUnmount() {
        clearTimeout(this.timeout);
        window.cancelAnimationFrame(this.rafHandle)
    }

    tick = () => {
        const { lastTick } = this
        const currentTime = performance.now()
        this.rafHandle = window.requestAnimationFrame(this.tick)

        if (currentTime - lastTick < 120) return

        this.lastTick = currentTime
        this.rotation.set(Animatable.getNumber(this.rotation) + 360 / 12)
    }

    render() {              
        const { children, width, height, indicatorColor } = this.props
        const { ready } = this.state;
        const dots: JSX.Element[] = []

        for (let i = 0, il = numberOfDots; i < il; i++) {
            const dot = (
                <Frame
                    key={i}
                    width={dotWidth}
                    height={dotHeight}
                    radius={dotWidth}
                    background={indicatorColor}
                    rotation={-i * (360 / numberOfDots)}
                    top={0}
                    opacity={1 - 0.05 * i}
                    originY={totalSize / 2 / dotHeight}
                />
            )
            dots.push(dot)
        }

        if (React.Children.count(this.props.children) !== 1) {
            return (
              <div style={hintStyle}>
                  Connect the purple outlet to the frame that represents the final state.
              </div>
            );
          }
  
        if (ready) {
            return (
              <Frame {...this.props} overflow="hidden">
                  {children}
              </Frame>
            );
          }
  
        return (
            <Frame
                width={totalSize}
                height={totalSize}
                parentSize={{ width, height }}
                rotation={this.rotation}
                background={null}
            >
                {dots}
            </Frame>
        )
    }
}

const hintStyle: React.CSSProperties = {
    height: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    color: "#8855FF",
    background: "rgba(136, 85, 255, 0.1)",
    overflow: "hidden",
    padding: 20
  };


const numberOfDots = 12
const totalSize = 20
const dotWidth = 2
const dotHeight = 5
