// @framer/9lpnaffm.exo-components/code/canvas.tsx
// WARNING: this file is auto generated, any changes will be lost
import { createDesignComponent, CanvasStore } from "framer"
const canvas = CanvasStore.shared(); // CANVAS_DATA;
export const Logo = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string,image?:string}>(canvas, "id_mYn0dvoj4", {image:"image"}, 281,69);
export const ModuleComplete = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_iV1Cb1zE7", {}, 20,20);
export const ModulePartialComplete25 = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_pWH0YR7tt", {}, 20,20);
export const ModulePartialComplete50 = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_UnRcCq7OL", {}, 20,20);
export const ModulePartialComplete75 = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_lnDDzU_CF", {}, 20,20);
export const SegmentCompleted = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string,displayText?:string}>(canvas, "id_XujN_dcsl", {displayText:"string"}, 115,14);
export const SegmentCurrent = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string,displayText?:string}>(canvas, "id_QxbDPJTHo", {displayText:"string"}, 115,14);
export const SegmentStateCompleted = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_WTLh_QdUO", {}, 12,12);
export const SegmentStateCurrent = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_AwswL2sSG", {}, 12,12);
export const SegmentStateUnstarted = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string}>(canvas, "id_aauyiYzF3", {}, 12,12);
export const SegmentUnstarted = createDesignComponent<{parentSize?:{width:number|string,height:number|string},width?:number|string,height?:number|string,displayText?:string}>(canvas, "id_nFyOGuZOm", {displayText:"string"}, 115,14);

export const colors = Object.freeze({
    /** #D4453B */
    "accent": "var(--token-2c0eeddf-1b06-43bc-863c-577102b585f9, rgb(212, 69, 59))",
    /** #B63F37 */
    "accent-hover": "var(--token-2c5485a2-8026-4a84-85a7-e1bd5c64d340, rgb(182, 63, 55))",
    /** #363940 */
    "body-bg": "var(--token-79bdced8-5310-4180-876b-565fc6cc6ee9, rgb(54, 57, 64))",
    /** #31333A */
    "body-bg-lighter": "var(--token-1f99af15-077a-4aaa-b264-6932cd8d5ab2, rgb(49, 51, 58))",
    /** #25282D */
    "body-bg-darker": "var(--token-15cd8612-582f-49bf-9410-f12eb8ca2ca2, rgb(37, 40, 45))",
    /** #EEEEEE */
    "body-text": "var(--token-d045f24e-e696-4f23-be26-5f315ac0e85a, rgb(238, 238, 238))",
    /** #999999 */
    "btn-bg": "var(--token-c6e8444c-3f50-4a2c-9ad7-d50fefc3f99d, rgb(153, 153, 153))",
    /** #666666 */
    "btn-bg-hover": "var(--token-6bb862a3-70d5-4614-ae8b-c01b5ee8a850, rgb(102, 102, 102))",
    /** #FFFFFF */
    "btn-text": "var(--token-6a22229c-7ef7-4de4-88d3-938aa1b8a78b, rgb(255, 255, 255))",
    /** #DDDDDD */
    "btn-reversed": "var(--token-b777d0f3-eb01-4144-afce-c6fd5bda9678, rgb(221, 221, 221))",
    /** #999999 */
    "btn-reversed-hover": "var(--token-5ad2a3b5-ddaa-44f9-8f3f-4bb31633c8f3, rgb(153, 153, 153))",
    /** #333333 */
    "btn-reversed-icon": "var(--token-2955335b-5012-41aa-985a-ce73beb624d4, rgb(51, 51, 51))",
    /** #2AB03A */
    "btn-primary-bg": "var(--token-784b3add-bb77-4dd3-82cc-9a80f6204f98, rgb(42, 176, 58))",
    /** #289D34 */
    "btn-primary-bg-hover": "var(--token-d9c08ba6-116b-4e9e-8504-0bff21cdd106, rgb(40, 157, 52))",
    /** #FFFFFF */
    "btn-primary-text": "var(--token-5e4437fb-824e-46e7-85d0-3b3e921e094a, rgb(255, 255, 255))",
    /** #2554AE */
    "btn-secondary-bg": "var(--token-a62ab184-6fb3-45bd-89ba-e9cdb8c7fe95, rgb(37, 84, 174))",
    /** #1E4BA2 */
    "btn-secondary-hover": "var(--token-2a424e8c-dfe5-4251-90a3-540fdbbcd0ef, rgb(30, 75, 162))",
    /** #FFFFFF */
    "btn-secondary-text": "var(--token-79e119a7-6da4-4fb7-929e-8ed744ddf643, rgb(255, 255, 255))",
    /** #FFFFFF */
    "content-bg": "var(--token-46304c82-aea3-4f9b-a028-07e8ee0b04dc, rgb(255, 255, 255))",
    /** #F7F7F7 */
    "content-bg-secondary": "var(--token-fe08b860-a8ad-4e79-bdd5-d573af82fc27, rgb(247, 247, 247))",
    /** rgba(255, 255, 255, 0) */
    "content-border": "var(--token-59d62f53-9ae1-4277-8693-4922aebb46a8, rgba(255, 255, 255, 0))",
    /** #DDDDDD */
    "content-separator": "var(--token-789d6b95-f8fe-404c-b35a-77253d63be84, rgb(221, 221, 221))",
    /** #111111 */
    "content-text": "var(--token-bf6ab502-16b5-49e4-92ed-2d9352a4366d, rgb(17, 17, 17))",
    /** #2053B4 */
    "current-segment-link": "var(--token-f266f7d0-2b7a-4cf4-8bd5-90cd10448142, rgb(32, 83, 180))",
    /** #555555 */
    "standard-icon": "var(--token-3bebc350-582d-41aa-91d9-e2f3086942ad, rgb(85, 85, 85))",
    /** #D4453B */
    "important-icon": "var(--token-25d4803b-cecf-4fda-aea3-fe84151d15c3, rgb(212, 69, 59))",
    /** #2554AE */
    "info-bg": "var(--token-394c4c8b-7f19-4468-a8c8-8d95ff94aa81, rgb(37, 84, 174))",
    /** #2AB03A */
    "info-bg-success": "var(--token-f546cc4c-7b7d-472c-967b-055d1460558b, rgb(42, 176, 58))",
    /** #D99F23 */
    "info-bg-warning": "var(--token-d61a210b-66e6-49c1-a23c-58bcbabb83ad, rgb(217, 159, 35))",
    /** #B3090F */
    "info-bg-danger": "var(--token-497852e5-72d6-40ae-86d4-abab03e5baad, rgb(179, 9, 15))",
    /** #DDDDDD */
    "info-bg-incomplete": "var(--token-c1a87526-ce9c-44d7-82f6-415788ab9f28, rgb(221, 221, 221))",
    /** #999999 */
    "info-bg-misc": "var(--token-5e690289-7090-4100-a4ab-a8f8626e0d36, rgb(153, 153, 153))",
    /** #FFFFFF */
    "info-text": "var(--token-75aa8b23-0015-4d5f-899c-ef9a8f87802f, rgb(255, 255, 255))",
    /** #44484D */
    "calendar-header-bg": "var(--token-b28fe66d-d12d-40c5-b5bd-26efba8dd887, rgb(68, 72, 77))",
    /** #FFFFFF */
    "calendar-header-text": "var(--token-197b2c94-483c-41d6-a87d-1f318985b1af, rgb(255, 255, 255))",
    /** #F9F9F9 */
    "calendar-body-bg": "var(--token-b2d71e90-efb9-4601-9e61-73dfa3ae717a, rgb(249, 249, 249))",
    /** #333333 */
    "calendar-body-text": "var(--token-dd169fd9-59a7-41a2-925a-a766940c819f, rgb(51, 51, 51))",
    /** #DEDEDE */
    "calendar-border": "var(--token-008d85df-5c85-4173-85ca-b0a86f26ae2c, rgb(222, 222, 222))",
    /** #2B2D33 */
    "sup-panel-bg": "var(--token-a36c682c-e88f-49ff-9719-ea73fe462067, rgb(43, 45, 51))",
    /** #24262B */
    "sup-panel-bg-darker": "var(--token-d9859c9c-5758-4366-82f4-8a5128e36680, rgb(36, 38, 43))",
    /** #363940 */
    "sup-panel-bg-lighter": "var(--token-0fe6e422-147a-48da-83bf-5343c6aadd47, rgb(54, 57, 64))",
    /** #FFFFFF */
    "sup-panel-text": "var(--token-0e053f4e-adfa-4f47-83eb-c596a161a58c, rgb(255, 255, 255))",
    /** #999999 */
    "sup-panel-outline-text": "var(--token-0c9c9515-df76-4d68-9d9e-4c332cfbd628, rgb(153, 153, 153))",
    /** #585858 */
    "content-page-subheader": "var(--token-b1c93e07-06db-42c9-9aa8-7c1491ff9a2a, rgb(88, 88, 88))",
    /** #FFFFFF */
    "input-bg": "var(--token-74d86a40-df9d-44df-a4e6-9b25a42b88fc, rgb(255, 255, 255))",
    /** #CCCCCC */
    "input-border": "var(--token-cb4c97ba-8756-4ec5-a143-b15817c62037, rgb(204, 204, 204))",
    /** #111111 */
    "input-text": "var(--token-4ef3961e-3b23-41a1-b7b6-7cd114454315, rgb(17, 17, 17))",
    /** #085CC9 */
    "input-active": "var(--token-bfb3b13c-6c53-4d30-8261-d6d50ec32b64, rgb(8, 92, 201))",
    /** #AAAAAA */
    "input-border-active": "var(--token-3ccdf14e-95a1-4720-906c-2a2e505a7739, rgb(170, 170, 170))",
    /** #085CC9 */
    "input-border-focused": "var(--token-9b83605e-42a0-429f-8829-2beec043f81a, rgb(8, 92, 201))",
    /** #B4B4B4 */
    "label": "var(--token-e9e211c3-35c9-4eb5-b4a9-0643f1c82661, rgb(180, 180, 180))",
    /** rgba(0, 0, 0, 0.5) */
    "modal-bg": "var(--token-600e4179-f024-42d8-a2bf-d6192df97499, rgba(0, 0, 0, 0.5))",
    /** #FFFFFF */
    "modal-content-bg": "var(--token-429e0336-b8ae-440a-984b-09fdfa3759e5, rgb(255, 255, 255))",
})
